src/shared/contexts/query.tsx

```diff
@@ -316,11 +361,105 @@ export const QueryProvider: FC = ({children}) => {
       headers['Authorization'] = `Token ${override.token}`
     }
 
-    const body = {
+    const body: RequestBody = {
       query,
       dialect: {annotations: ['group', 'datatype', 'default']},
     }
 
+    if (mechanism !== OverrideMechanism.Injection) {
+      const options: Record<string, any> = {}
+
+      if (Object.keys(override?.vars ?? {}).length) {
+        options.v = override.vars
+      }
+      if (Object.keys(override?.params ?? {}).length) {
+        options.params = override.params
+      }
+      if (Object.keys(override?.task ?? {}).length) {
+        options.task = override.task
+      }
+
+      const optionTexts = Object.entries(options)
+        .map(([k, v]) => {
+          const vals = Object.entries(v).map(([_k, _v]) => `  ${_k}: ${_v}`)
+          return `option ${k} =  {${vals.join(',\n')}}`
+        })
+        .join('\n\n')
+
+      const queryAST = parse(query)
+      let optionAST = parse(optionTexts)
+
+      // only run this if the query need a windowPeriod
+      if (
+        find(
+          queryAST,
+          node =>
+            node?.type === 'MemberExpression' &&
+            node?.object?.name === 'v' &&
+            node?.property?.name === 'windowPeriod'
+        ).length
+      ) {
+        try {
+          const _optionAST = JSON.parse(JSON.stringify(optionAST))
+          // make sure there's a variable in there named windowPeriod so later logic doesnt bail
+          find(
+            _optionAST,
+            node =>
+              node?.type === 'OptionStatement' &&
+              node?.assignment?.id?.name === 'v'
+          ).forEach(node => {
+            if (
+              find(
+                node,
+                n => n.type === 'Property' && n?.key?.name === 'windowPeriod'
+              ).length
+            ) {
+              return
+            }
+
+            node.assignment.init.properties.push({
+              type: 'Property',
+              key: {
+                type: 'Identifier',
+                name: 'windowPeriod',
+              },
+              value: {
+                type: 'DurationLiteral',
+                values: [{magnitude: FALLBACK_WINDOW_PERIOD, unit: 'ms'}],
```

Comment:
On fallback (a.k.a. cannot modify the AST...such as the complex queries), can we try a tiered fallback?
1. using the windowPeriod from the presets (`SELECTABLE_TIME_RANGES`)
2. then the `FALLBACK_WINDOW_PERIOD`

Reply:
i do  not know what you mean

Reply:
are you talking about this that's already in there? https://github.com/influxdata/ui/blob/master/src/shared/contexts/query.tsx#L158

that wont work for the bug i'm working on as it fails here https://github.com/influxdata/ui/blob/master/src/shared/contexts/query.tsx#L117, meaning we aren't getting to any valid ranges to look up against the `SELECTABLE_TIME_RANGES`,  and i can't find any equivalent functionality in the old data explorer code path that utilizes this fallback differently

maybe i'm missing something?

Reply:
offlined: this was a feature request and out of scope